diff --git a/test/functionalities/register/TestRegisters.py b/test/functionalities/register/TestRegisters.py
index b78a254..6088dbe 100644
--- a/test/functionalities/register/TestRegisters.py
+++ b/test/functionalities/register/TestRegisters.py
@@ -28,6 +29,7 @@ class RegisterCommandsTestCase(TestBase):
         self.buildDefault()
         self.register_commands()
 
+    @unittest2.expectedFailure
     def test_fp_register_write(self):
         """Test commands that write to registers, in particular floating-point registers."""
         if not self.getArchitecture() in ['amd64', 'i386', 'x86_64']:
