diff --git a/packages/Python/lldbsuite/test/functionalities/thread/jump/TestThreadJump.py b/packages/Python/lldbsuite/test/functionalities/thread/jump/TestThreadJump.py
index 768e2fe..274d33a 100644
--- a/packages/Python/lldbsuite/test/functionalities/thread/jump/TestThreadJump.py
+++ b/packages/Python/lldbsuite/test/functionalities/thread/jump/TestThreadJump.py
@@ -15,6 +15,7 @@ class ThreadJumpTestCase(TestBase):
 
     mydir = TestBase.compute_mydir(__file__)
 
+    @skipIfLinux
     def test(self):
         """Test thread jump handling."""
         self.build(dictionary=self.getBuildFlags())
