diff --git a/test/functionalities/process_launch/TestProcessLaunch.py b/test/functionalities/process_launch/TestProcessLaunch.py
index 6819c36..92b678e 100644
--- a/test/functionalities/process_launch/TestProcessLaunch.py
+++ b/test/functionalities/process_launch/TestProcessLaunch.py
@@ -26,6 +26,7 @@ class ProcessLaunchTestCase(TestBase):
         self.process_io_test ()
 
     @dwarf_test
+    @unittest2.expectedFailure
     def test_io_with_dwarf (self):
         """Test that process launch I/O redirection flags work properly."""
         self.buildDwarf ()
