diff --git a/test/functionalities/thread/jump/TestThreadJump.py b/test/functionalities/thread/jump/TestThreadJump.py
index d1c279f..3a28eff 100644
--- a/test/functionalities/thread/jump/TestThreadJump.py
+++ b/test/functionalities/thread/jump/TestThreadJump.py
@@ -20,6 +20,7 @@ class ThreadJumpTestCase(TestBase):
         self.thread_jump_test()
 
     @dwarf_test
+    @skipIfi386
     def test_with_dwarf(self):
         """Test thread jump handling."""
         self.buildDwarf(dictionary=self.getBuildFlags())
