diff --git a/test/python_api/lldbutil/process/TestPrintStackTraces.py b/test/python_api/lldbutil/process/TestPrintStackTraces.py
index ce0293b..360fdcc 100644
--- a/test/python_api/lldbutil/process/TestPrintStackTraces.py
+++ b/test/python_api/lldbutil/process/TestPrintStackTraces.py
@@ -18,7 +18,7 @@ class ThreadsStackTracesTestCase(TestBase):
         # Find the line number to break inside main().
         self.line = line_number('main.cpp', '// Set break point at this line.')
 
-    @expectedFailureAll("llvm.org/pr23043", ["linux"], archs=["i386"]) # We are unable to produce a backtrace of the main thread when the thread is blocked in fgets
+    @expectedFailureAll("llvm.org/pr23043", ["linux"]) # We are unable to produce a backtrace of the main thread when the thread is blocked in fgets
     @python_api_test
     def test_stack_traces(self):
         """Test SBprocess and SBThread APIs with printing of the stack traces."""
