diff --git a/packages/Python/lldbsuite/test/functionalities/signal/TestSendSignal.py b/packages/Python/lldbsuite/test/functionalities/signal/TestSendSignal.py
index 971b82a..20af652 100644
--- a/packages/Python/lldbsuite/test/functionalities/signal/TestSendSignal.py
+++ b/packages/Python/lldbsuite/test/functionalities/signal/TestSendSignal.py
@@ -22,6 +22,7 @@ class SendSignalTestCase(TestBase):
 
     @expectedFailureFreeBSD("llvm.org/pr23318: does not report running state")
     @skipIfWindows # Windows does not support signals
+    @unittest2.expectedFailure
     def test_with_run_command(self):
         """Test that lldb command 'process signal SIGUSR1' sends a signal to the inferior process."""
         self.build()
