diff --git a/test/tools/lldb-server/commandline/TestStubSetSID.py b/test/tools/lldb-server/commandline/TestStubSetSID.py
index 3cea264..cad0109 100644
--- a/test/tools/lldb-server/commandline/TestStubSetSID.py
+++ b/test/tools/lldb-server/commandline/TestStubSetSID.py
@@ -61,6 +61,7 @@ class TestStubSetSIDTestCase(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @skipIfRemote # --setsid not used on remote platform and currently it is also impossible to get the sid of lldb-platform running on a remote target
+    @unittest2.expectedFailure() # Setsid is broken on ds2
     def test_sid_is_different_with_setsid_llgs(self):
         self.init_llgs_test()
         self.set_inferior_startup_launch()
@@ -75,6 +76,7 @@ class TestStubSetSIDTestCase(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @skipIfRemote # --setsid not used on remote platform and currently it is also impossible to get the sid of lldb-platform running on a remote target
+    @unittest2.expectedFailure() # Setsid is broken on ds2
     def test_sid_is_different_with_S_llgs(self):
         self.init_llgs_test()
         self.set_inferior_startup_launch()
