diff --git a/test/functionalities/signal/raise/TestRaise.py b/test/functionalities/signal/raise/TestRaise.py
index 3005ee8..a5cb5fe 100644
--- a/test/functionalities/signal/raise/TestRaise.py
+++ b/test/functionalities/signal/raise/TestRaise.py
@@ -24,6 +24,7 @@ class RaiseTestCase(TestBase):
     @skipIfWindows # signals do not exist on Windows
     @dwarf_test
     @expectedFailureDarwin("llvm.org/pr23610") # process doesn't stop at a breakpoint on the third launch
+    @unittest2.expectedFailure
     def test_sigstop_with_dwarf(self):
         self.buildDwarf()
         self.signal_test('SIGSTOP', False)
@@ -33,6 +34,7 @@ class RaiseTestCase(TestBase):
     @dwarf_test
     @skipIfDarwin # darwin does not support real time signals
     @skipIfTargetAndroid()
+    @unittest2.expectedFailure
     def test_sigsigrtmin_with_dwarf(self):
         self.buildDwarf()
         self.signal_test('SIGRTMIN', True)
