diff --git a/test/functionalities/thread/thread_exit/TestThreadExit.py b/test/functionalities/thread/thread_exit/TestThreadExit.py
index 7b8ed18..0547324 100644
--- a/test/functionalities/thread/thread_exit/TestThreadExit.py
+++ b/test/functionalities/thread/thread_exit/TestThreadExit.py
@@ -22,6 +22,7 @@ class ThreadExitTestCase(TestBase):
 
     @expectedFailureDarwin("llvm.org/pr15824") # thread states not properly maintained
     @expectedFailureFreeBSD("llvm.org/pr18190") # thread states not properly maintained
+    @skipIfLinux
     @dwarf_test
     def test_with_dwarf(self):
         """Test thread exit handling."""
