diff --git a/test/expression_command/timeout/TestCallWithTimeout.py b/test/expression_command/timeout/TestCallWithTimeout.py
index 4cc5577..97ada28 100644
--- a/test/expression_command/timeout/TestCallWithTimeout.py
+++ b/test/expression_command/timeout/TestCallWithTimeout.py
@@ -30,6 +30,7 @@ class ExprCommandWithTimeoutsTestCase(TestBase):
     @expectedFailureFreeBSD("llvm.org/pr19605") # fails on buildbot
     @expectedFlakeyLinux("llvm.org/pr20275")
     @dwarf_test
+    @unittest2.expectedFailure
     def test_with_dwarf(self):
         """Test calling std::String member function."""
         self.buildDwarf()
