diff --git a/packages/Python/lldbsuite/test/functionalities/return-value/TestReturnValue.py b/packages/Python/lldbsuite/test/functionalities/return-value/TestReturnValue.py
index 246eb5c..790ddb7 100644
--- a/packages/Python/lldbsuite/test/functionalities/return-value/TestReturnValue.py
+++ b/packages/Python/lldbsuite/test/functionalities/return-value/TestReturnValue.py
@@ -21,6 +21,7 @@ class ReturnValueTestCase(TestBase):
     @expectedFailureAll(bugnumber="llvm.org/pr25785", hostoslist=["windows"], compiler="gcc", archs=["i386"], triple='.*-android')
     @expectedFailureWindows("llvm.org/pr24778")
     @add_test_categories(['pyapi'])
+    @skipIfLinux
     def test_with_python(self):
         """Test getting return values from stepping out."""
         self.build()
