diff --git a/test/tools/lldb-server/TestLldbGdbServer.py b/test/tools/lldb-server/TestLldbGdbServer.py
index 03dcc84..2c40d9b 100644
--- a/test/tools/lldb-server/TestLldbGdbServer.py
+++ b/test/tools/lldb-server/TestLldbGdbServer.py
@@ -682,6 +682,7 @@ class LldbGdbServerTestCase(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @dwarf_test
+    @unittest2.expectedFailure
     def test_p_returns_correct_data_size_for_each_qRegisterInfo_launch_llgs_dwarf(self):
         self.init_llgs_test()
         self.buildDwarf()
@@ -698,6 +699,7 @@ class LldbGdbServerTestCase(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @dwarf_test
+    @unittest2.expectedFailure
     def test_p_returns_correct_data_size_for_each_qRegisterInfo_attach_llgs_dwarf(self):
         self.init_llgs_test()
         self.buildDwarf()
@@ -743,6 +745,7 @@ class LldbGdbServerTestCase(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @dwarf_test
+    @unittest2.expectedFailure
     def test_Hg_switches_to_3_threads_launch_llgs_dwarf(self):
         self.init_llgs_test()
         self.buildDwarf()
@@ -759,6 +762,7 @@ class LldbGdbServerTestCase(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @dwarf_test
+    @unittest2.expectedFailure
     def test_Hg_switches_to_3_threads_attach_llgs_dwarf(self):
         self.init_llgs_test()
         self.buildDwarf()
@@ -879,6 +883,7 @@ class LldbGdbServerTestCase(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @dwarf_test
+    @unittest2.expectedFailure() #not really a failure, but super flakey
     def test_Hc_then_Csignal_signals_correct_thread_launch_llgs_dwarf(self):
         self.init_llgs_test()
         self.buildDwarf()
@@ -1521,6 +1526,7 @@ class LldbGdbServerTestCase(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @dwarf_test
+    @unittest2.expectedFailure
     def test_P_and_p_thread_suffix_work_llgs_dwarf(self):
         self.init_llgs_test()
         self.buildDwarf()
