diff --git a/test/functionalities/thread/exit_during_step/TestExitDuringStep.py b/test/functionalities/thread/exit_during_step/TestExitDuringStep.py
index f04215c..a22c59e 100644
--- a/test/functionalities/thread/exit_during_step/TestExitDuringStep.py
+++ b/test/functionalities/thread/exit_during_step/TestExitDuringStep.py
@@ -52,6 +52,7 @@ class ExitDuringStepTestCase(TestBase):
 
     @skipIfFreeBSD # llvm.org/pr21411: test is hanging
     @dwarf_test
+    @unittest2.expectedFailure
     def test_with_dwarf(self):
         """Test thread exit during step handling."""
         self.buildDwarf(dictionary=self.getBuildFlags())
