diff --git a/test/lang/c/register_variables/TestRegisterVariables.py b/test/lang/c/register_variables/TestRegisterVariables.py
index 101e5c1..0166f04 100644
--- a/test/lang/c/register_variables/TestRegisterVariables.py
+++ b/test/lang/c/register_variables/TestRegisterVariables.py
@@ -19,7 +19,7 @@ class RegisterVariableTestCase(TestBase):
     @dwarf_test
     @expectedFailureClang(None, ['<', '3.5'])
-    @expectedFailureGcc(None, ['is', '4.8.2'])
+    @expectedFailureGcc(None, ['>=', '4.8.2'])
     def test_with_dwarf_and_run_command(self):
         """Test expressions on register values."""
         self.buildDwarf()
