diff --git a/test/python_api/process/io/TestProcessIO.py b/test/python_api/process/io/TestProcessIO.py
index 66ac0bd..86c5eb9 100644
--- a/test/python_api/process/io/TestProcessIO.py
+++ b/test/python_api/process/io/TestProcessIO.py
@@ -21,6 +21,7 @@ class ProcessIOTestCase(TestBase):
     @skipIfWindows # stdio manipulation unsupported on Windows
     @python_api_test
     @dwarf_test
+    @unittest2.expectedFailure 
     def test_stdin_by_api_with_dwarf(self):
         """Exercise SBProcess.PutSTDIN()."""
         self.buildDwarf()
@@ -37,6 +38,7 @@ class ProcessIOTestCase(TestBase):
     @skipIfWindows # stdio manipulation unsupported on Windows
     @python_api_test
     @dwarf_test
+    @unittest2.expectedFailure 
     def test_stdin_redirection_with_dwarf(self):
         """Exercise SBLaunchInfo::AddOpenFileAction() for STDIN without specifying STDOUT or STDERR."""
         self.buildDwarf()
@@ -53,6 +55,7 @@ class ProcessIOTestCase(TestBase):
     @skipIfWindows # stdio manipulation unsupported on Windows
     @python_api_test
     @dwarf_test
+    @unittest2.expectedFailure 
     def test_stdout_redirection_with_dwarf(self):
         """Exercise SBLaunchInfo::AddOpenFileAction() for STDOUT without specifying STDIN or STDERR."""
         self.buildDwarf()
@@ -69,6 +72,7 @@ class ProcessIOTestCase(TestBase):
     @skipIfWindows # stdio manipulation unsupported on Windows
     @python_api_test
     @dwarf_test
+    @unittest2.expectedFailure
     def test_stderr_redirection_with_dwarf(self):
         """Exercise SBLaunchInfo::AddOpenFileAction() for STDERR without specifying STDIN or STDOUT."""
         self.buildDwarf()
@@ -85,6 +89,7 @@ class ProcessIOTestCase(TestBase):
     @skipIfWindows # stdio manipulation unsupported on Windows
     @python_api_test
     @dwarf_test
+    @unittest2.expectedFailure 
     def test_stdout_stderr_redirection_with_dwarf(self):
         """Exercise SBLaunchInfo::AddOpenFileAction() for STDOUT and STDERR without redirecting STDIN."""
         self.buildDwarf()
