diff --git a/test/functionalities/command_script/import/TestImport.py b/test/functionalities/command_script/import/TestImport.py
index 393285b..ccdfccf 100644
--- a/test/functionalities/command_script/import/TestImport.py
+++ b/test/functionalities/command_script/import/TestImport.py
@@ -10,6 +10,7 @@ class ImportTestCase(TestBase):
     mydir = TestBase.compute_mydir(__file__)
 
     @python_api_test
+    @unittest2.expectedFailure
     def test_import_command(self):
         """Import some Python scripts by path and test them"""
         self.run_test()
