diff --git a/test/settings/quoting/TestQuoting.py b/test/settings/quoting/TestQuoting.py
index c65971f..be2d246 100644
--- a/test/settings/quoting/TestQuoting.py
+++ b/test/settings/quoting/TestQuoting.py
@@ -16,45 +16,59 @@ class SettingsCommandTestCase(TestBase):
         """Cleanup the test byproducts."""
         cls.RemoveTempFile("stdout.txt")
 
+    @unittest2.expectedFailure
     def test_no_quote(self):
         self.do_test_args("a b c", "a\0b\0c\0")
 
+    @unittest2.expectedFailure
     def test_single_quote(self):
         self.do_test_args("'a b c'", "a b c\0")
 
+    @unittest2.expectedFailure
     def test_double_quote(self):
         self.do_test_args('"a b c"', "a b c\0")
 
+    @unittest2.expectedFailure
     def test_double_quote(self):
         self.do_test_args('"a b c"', 'a b c\0')
 
+    @unittest2.expectedFailure
     def test_single_quote_escape(self):
         self.do_test_args("'a b\\' c", "a b\\\0c\0")
 
+    @unittest2.expectedFailure
     def test_double_quote_escape(self):
         self.do_test_args('"a b\\" c"', 'a b" c\0')
 
+    @unittest2.expectedFailure
     def test_double_quote_escape(self):
         self.do_test_args('"a b\\" c"', 'a b" c\0')
 
+    @unittest2.expectedFailure
     def test_double_quote_escape2(self):
         self.do_test_args('"a b\\\\" c', 'a b\\\0c\0')
 
+    @unittest2.expectedFailure
     def test_double_quote_escape2(self):
         self.do_test_args('"a b\\\\" c', 'a b\\\0c\0')
 
+    @unittest2.expectedFailure
     def test_single_in_double(self):
         self.do_test_args('"a\'b"', "a'b\0")
 
+    @unittest2.expectedFailure
     def test_double_in_single(self):
         self.do_test_args("'a\"b'", 'a"b\0')
 
+    @unittest2.expectedFailure
     def test_combined(self):
         self.do_test_args('"a b"c\'d e\'', 'a bcd e\0')
 
+    @unittest2.expectedFailure
     def test_bare_single(self):
         self.do_test_args("a\\'b", "a'b\0")
 
+    @unittest2.expectedFailure
     def test_bare_double(self):
         self.do_test_args('a\\"b', 'a"b\0')
 
