diff --git a/test/python_api/event/TestEvents.py b/test/python_api/event/TestEvents.py
index ec8ad18..2db1fa9 100644
--- a/test/python_api/event/TestEvents.py
+++ b/test/python_api/event/TestEvents.py
@@ -56,6 +56,7 @@ class EventAPITestCase(TestBase):
     @python_api_test
     @dwarf_test
     @expectedFailureLinux("llvm.org/pr23617") # Flaky, fails ~1/10 cases
+    @unittest2.expectedFailure #should already be an expected failure? 
     def test_add_listener_to_broadcaster_with_dwarf(self):
         """Exercise some SBBroadcaster APIs."""
         self.buildDwarf()
