diff --git a/test/lang/c/shared_lib_stripped_symbols/TestSharedLibStrippedSymbols.py b/test/lang/c/shared_lib_stripped_symbols/TestSharedLibStrippedSymbols.py
index 0987e78..ee9a1cf 100644
--- a/test/lang/c/shared_lib_stripped_symbols/TestSharedLibStrippedSymbols.py
+++ b/test/lang/c/shared_lib_stripped_symbols/TestSharedLibStrippedSymbols.py
@@ -16,6 +16,7 @@ class SharedLibStrippedTestCase(TestBase):
         self.expr()
 
     @dwarf_test
+    @unittest2.expectedFailure
     def test_expr_with_dwarf(self):
         """Test that types work when defined in a shared library and forward-declared in the main executable"""
         self.buildDwarf()
