diff --git a/test/python_api/thread/TestThreadAPI.py b/test/python_api/thread/TestThreadAPI.py
index 706919e..c5727d2 100644
--- a/test/python_api/thread/TestThreadAPI.py
+++ b/test/python_api/thread/TestThreadAPI.py
@@ -77,6 +77,7 @@ class ThreadAPITestCase(TestBase):
     @expectedFailureFreeBSD # llvm.org/pr20476
     @python_api_test
     @dwarf_test
+    @unittest2.expectedFailure 
     def test_step_out_of_malloc_into_function_b_with_dwarf(self):
         """Test Python SBThread.StepOut() API to step out of a malloc call where the call site is at function b()."""
         # We build a different executable than the default buildDwarf() does.
