diff --git a/test/types/TestFloatTypes.py b/test/types/TestFloatTypes.py
index 7f403fe..fd53118 100644
--- a/test/types/TestFloatTypes.py
+++ b/test/types/TestFloatTypes.py
@@ -32,6 +32,7 @@ class FloatTypesTestCase(AbstractBase.GenericTester):
         self.build_and_run('float.cpp', set(['float']), bc=True)
 
     @dwarf_test
+    @unittest2.expectedFailure 
     def test_float_type_with_dwarf(self):
         """Test that float-type variables are displayed correctly."""
         self.build_and_run('float.cpp', set(['float']), dsym=False)
@@ -50,6 +51,7 @@ class FloatTypesTestCase(AbstractBase.GenericTester):
         self.build_and_run('double.cpp', set(['double']), bc=True)
 
     @dwarf_test
+    @unittest2.expectedFailure 
     def test_double_type_with_dwarf(self):
         """Test that double-type variables are displayed correctly."""
         self.build_and_run('double.cpp', set(['double']), dsym=False)
diff --git a/test/types/TestFloatTypesExpr.py b/test/types/TestFloatTypesExpr.py
index 33ff0c5..f155750 100644
--- a/test/types/TestFloatTypesExpr.py
+++ b/test/types/TestFloatTypesExpr.py
@@ -35,6 +35,7 @@ class FloatTypesExprTestCase(AbstractBase.GenericTester):
         self.build_and_run_expr('float.cpp', set(['float']), bc=True)
 
     @dwarf_test
+    @unittest2.expectedFailure 
     def test_float_type_with_dwarf(self):
         """Test that float-type variable expressions are evaluated correctly."""
         self.build_and_run_expr('float.cpp', set(['float']), dsym=False)
@@ -52,6 +53,7 @@ class FloatTypesExprTestCase(AbstractBase.GenericTester):
         self.build_and_run_expr('double.cpp', set(['double']), bc=True)
 
     @dwarf_test
+    @unittest2.expectedFailure 
     def test_double_type_with_dwarf(self):
         """Test that double-type variable expressions are evaluated correctly."""
         self.build_and_run_expr('double.cpp', set(['double']), dsym=False)
