diff --git a/test/python_api/value_var_update/TestValueVarUpdate.py b/test/python_api/value_var_update/TestValueVarUpdate.py
index f3aa296..e5277ff 100644
--- a/test/python_api/value_var_update/TestValueVarUpdate.py
+++ b/test/python_api/value_var_update/TestValueVarUpdate.py
@@ -22,6 +22,7 @@ class HelloWorldTestCase(TestBase):
     @expectedFailureFreeBSD("llvm.org/pr21620 GetValueDidChange() wrong")
     @python_api_test
     @dwarf_test
+    @unittest2.expectedFailure 
     def test_with_dwarf_and_process_launch_api(self):
         """Test SBValue::GetValueDidChange"""
         self.buildDwarf(dictionary=self.d)
