diff --git a/packages/Python/lldbsuite/test/lang/c/stepping/TestStepAndBreakpoints.py b/packages/Python/lldbsuite/test/lang/c/stepping/TestStepAndBreakpoints.py
index c7a3de4..d91cdf9 100644
--- a/packages/Python/lldbsuite/test/lang/c/stepping/TestStepAndBreakpoints.py
+++ b/packages/Python/lldbsuite/test/lang/c/stepping/TestStepAndBreakpoints.py
@@ -23,7 +23,6 @@ class TestCStepping(TestBase):
         self.main_source = "main.c"
 
     @expectedFailureFreeBSD('llvm.org/pr17932')
-    @expectedFailureLinux # llvm.org/pr14437
     @expectedFailureWindows("llvm.org/pr24777")
     @add_test_categories(['pyapi'])
     def test_and_python_api(self):
