diff --git a/packages/Python/lldbsuite/test/functionalities/signal/raise/TestRaise.py b/packages/Python/lldbsuite/test/functionalities/signal/raise/TestRaise.py
index 39e7753..b6aef3c 100644
--- a/packages/Python/lldbsuite/test/functionalities/signal/raise/TestRaise.py
+++ b/packages/Python/lldbsuite/test/functionalities/signal/raise/TestRaise.py
@@ -148,7 +148,6 @@ class RaiseTestCase(TestBase):
         # reset signal handling to default
         self.set_handle(signal, default_pass, default_stop, default_notify)
 
-    @expectedFailureLinux("llvm.org/pr24530") # the signal the inferior generates gets lost
     @expectedFailureDarwin("llvm.org/pr24530") # the signal the inferior generates gets lost
     def test_restart_bug(self):
         """Test that we catch a signal in the edge case where the process receives it while we are
