From 0aedc033f6418bbd086c62f34b0f78d4d31cdcee Mon Sep 17 00:00:00 2001
From: Corentin Derbois <cderbois@gmail.com>
Date: Fri, 6 May 2016 03:17:48 -0700
Subject: [PATCH] toto

---
 test/dotest.py | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/test/dotest.py b/test/dotest.py
index 4969304..571062e 100755
--- a/test/dotest.py
+++ b/test/dotest.py
@@ -1038,8 +1038,8 @@ def setupSysPath():
         
         if not lldbPythonDir:
             if platform.system() == "Darwin":
-                python_resource_dir = ['LLDB.framework', 'Resources', 'Python']
-                outputPaths = getXcodeOutputPaths()
+                python_resource_dir = os.path.join('LLDB.framework', 'Resources', 'Python')
+                outputPaths = getXcodeOutputPaths(lldbRootDirectory)
                 for outputPath in outputPaths:
                     candidatePath = os.path.join(outputPath, python_resource_dir)
                     if os.path.isfile(os.path.join(candidatePath, init_in_python_dir)):
@@ -1047,8 +1047,7 @@ def setupSysPath():
                         break
 
                 if not lldbPythonDir:
-                    print 'This script requires lldb.py to be in either ' + dbgPath + ',',
-                    print relPath + ', or ' + baiPath + '. Some tests might fail.'
+                    print 'lldb not found some test could not work {}'.format(getXcodeOutputPaths(lldbRootDirectory))
             else:
                 print "Unable to load lldb extension module.  Possible reasons for this include:"
                 print "  1) LLDB was built with LLDB_DISABLE_PYTHON=1"
-- 
2.5.4 (Apple Git-61)

