diff --git a/test/plugins/builder_base.py b/test/plugins/builder_base.py
index 07df974..e9e8e12 100644
--- a/test/plugins/builder_base.py
+++ b/test/plugins/builder_base.py
@@ -22,7 +22,8 @@ def getArchitecture():
 
 def getCompiler():
     """Returns the compiler in effect the test suite is running with."""
-    return os.environ["CC"] if "CC" in os.environ else "clang"
+    compiler = os.environ.get("CC", "clang")
+    return os.path.realpath(compiler)
 
 def getArchFlag():
     """Returns the flag required to specify the arch"""
