diff --git a/test/lldbtest.py b/test/lldbtest.py
index 672f445..55ec945 100644
--- a/test/lldbtest.py
+++ b/test/lldbtest.py
@@ -865,10 +865,10 @@ def skipUnlessDarwin(func):
 
 def getPlatform():
     """Returns the target platform which the tests are running on."""
-    platform = lldb.DBG.GetSelectedPlatform().GetTriple().split('-')[2]
-    if platform.startswith('freebsd'):
-        platform = 'freebsd'
-    return platform
+    # For some reason this likes to crash, so let's hack it
+    if sys.platform == 'darwin':
+        return 'darwin'
+    return 'linux'
 
 def getHostPlatform():
     """Returns the host platform running the test suite."""
