diff --git a/packages/Python/lldbsuite/test/lang/c/register_variables/TestRegisterVariables.py b/packages/Python/lldbsuite/test/lang/c/register_variables/TestRegisterVariables.py
index 7ef1f24..c892bed 100644
--- a/packages/Python/lldbsuite/test/lang/c/register_variables/TestRegisterVariables.py
+++ b/packages/Python/lldbsuite/test/lang/c/register_variables/TestRegisterVariables.py
@@ -15,7 +15,7 @@ class RegisterVariableTestCase(TestBase):
 
     @expectedFailureAll(oslist=['macosx'], compiler='clang', compiler_version=['<', '7.0.0'], debug_info="dsym")
     @expectedFailureClang(None, ['<', '3.5'])
-    @expectedFailureGcc(None, ['is', '4.8.2'])
+    @expectedFailureGcc(None, ['>=', '4.8.2'])
     def test_and_run_command(self):
         """Test expressions on register values."""
         self.build()
