diff --git a/packages/Python/lldbsuite/test/functionalities/process_attach/attach_denied/TestAttachDenied.py b/packages/Python/lldbsuite/test/functionalities/process_attach/attach_denied/TestAttachDenied.py
index ed9d58f..5793414 100644
--- a/packages/Python/lldbsuite/test/functionalities/process_attach/attach_denied/TestAttachDenied.py
+++ b/packages/Python/lldbsuite/test/functionalities/process_attach/attach_denied/TestAttachDenied.py
@@ -25,6 +25,7 @@ class AttachDeniedTestCase(TestBase):
 
     @skipIfWindows
     @skipIfiOSSimulator
+    @unittest2.expectedFailure
     def test_attach_to_process_by_id_denied(self):
         """Test attach by process id denied"""
         self.build()
