diff --git a/packages/Python/lldbsuite/test/python_api/process/io/TestProcessIO.py b/packages/Python/lldbsuite/test/python_api/process/io/TestProcessIO.py
index 2944ee7..6b1d49c 100644
--- a/packages/Python/lldbsuite/test/python_api/process/io/TestProcessIO.py
+++ b/packages/Python/lldbsuite/test/python_api/process/io/TestProcessIO.py
@@ -29,6 +29,7 @@ class ProcessIOTestCase(TestBase):
 
     @skipIfWindows # stdio manipulation unsupported on Windows
     @add_test_categories(['pyapi'])
+    @unittest2.expectedFailure
     def test_stdin_by_api(self):
         """Exercise SBProcess.PutSTDIN()."""
         self.build()
@@ -50,6 +51,7 @@ class ProcessIOTestCase(TestBase):
 
     @skipIfWindows # stdio manipulation unsupported on Windows
     @add_test_categories(['pyapi'])
+    @unittest2.expectedFailure
     def test_stdout_redirection(self):
         """Exercise SBLaunchInfo::AddOpenFileAction() for STDOUT without specifying STDIN or STDERR."""
         self.build()
@@ -62,6 +64,7 @@ class ProcessIOTestCase(TestBase):
 
     @skipIfWindows # stdio manipulation unsupported on Windows
     @add_test_categories(['pyapi'])
+    @unittest2.expectedFailure
     def test_stderr_redirection(self):
         """Exercise SBLaunchInfo::AddOpenFileAction() for STDERR without specifying STDIN or STDOUT."""
         self.build()
@@ -74,6 +77,7 @@ class ProcessIOTestCase(TestBase):
 
     @skipIfWindows # stdio manipulation unsupported on Windows
     @add_test_categories(['pyapi'])
+    @unittest2.expectedFailure
     def test_stdout_stderr_redirection(self):
         """Exercise SBLaunchInfo::AddOpenFileAction() for STDOUT and STDERR without redirecting STDIN."""
         self.build()
