diff --git a/packages/Python/lldbsuite/test/python_api/hello_world/TestHelloWorld.py b/packages/Python/lldbsuite/test/python_api/hello_world/TestHelloWorld.py
index 47c3ba1..f3c95da 100644
--- a/packages/Python/lldbsuite/test/python_api/hello_world/TestHelloWorld.py
+++ b/packages/Python/lldbsuite/test/python_api/hello_world/TestHelloWorld.py
@@ -77,6 +77,7 @@ class HelloWorldTestCase(TestBase):
     @expectedFailureWindows("llvm.org/pr24600")
     @expectedFailurei386("llvm.org/pr25338")
     @skipIfiOSSimulator
+    @expectedFailureLinux
     def test_with_attach_to_process_with_id_api(self):
         """Create target, spawn a process, and attach to it with process id."""
         self.build(dictionary=self.d)
@@ -107,6 +108,7 @@ class HelloWorldTestCase(TestBase):
     @expectedFailureWindows("llvm.org/pr24600")
     @expectedFailurei386("llvm.org/pr25338")
     @skipIfiOSSimulator
+    @skipIfRemote
     def test_with_attach_to_process_with_name_api(self):
         """Create target, spawn a process, and attach to it with process name."""
         self.build(dictionary=self.d)
