diff --git a/packages/Python/lldbsuite/test/settings/quoting/TestQuoting.py b/packages/Python/lldbsuite/test/settings/quoting/TestQuoting.py
index 878fc8a..86a0e72 100644
--- a/packages/Python/lldbsuite/test/settings/quoting/TestQuoting.py
+++ b/packages/Python/lldbsuite/test/settings/quoting/TestQuoting.py
@@ -20,52 +20,63 @@ class SettingsCommandTestCase(TestBase):
         cls.RemoveTempFile("stdout.txt")
 
     @no_debug_info_test
+    @unittest2.expectedFailure
     def test_no_quote(self):
         self.do_test_args("a b c", "a\0b\0c\0")
 
     @expectedFailureWindows("http://llvm.org/pr24557")
     @no_debug_info_test
+    @unittest2.expectedFailure
     def test_single_quote(self):
         self.do_test_args("'a b c'", "a b c\0")
 
     @no_debug_info_test
+    @unittest2.expectedFailure
     def test_double_quote(self):
         self.do_test_args('"a b c"', "a b c\0")
 
     @expectedFailureWindows("http://llvm.org/pr24557")
     @no_debug_info_test
+    @unittest2.expectedFailure
     def test_single_quote_escape(self):
         self.do_test_args("'a b\\' c", "a b\\\0c\0")
 
     @expectedFailureWindows("http://llvm.org/pr24557")
     @no_debug_info_test
+    @unittest2.expectedFailure
     def test_double_quote_escape(self):
         self.do_test_args('"a b\\" c"', 'a b" c\0')
 
     @expectedFailureWindows("http://llvm.org/pr24557")
     @no_debug_info_test
+    @unittest2.expectedFailure
     def test_double_quote_escape2(self):
         self.do_test_args('"a b\\\\" c', 'a b\\\0c\0')
 
     @no_debug_info_test
+    @unittest2.expectedFailure
     def test_single_in_double(self):
         self.do_test_args('"a\'b"', "a'b\0")
 
     @expectedFailureWindows("http://llvm.org/pr24557")
     @no_debug_info_test
+    @unittest2.expectedFailure
     def test_double_in_single(self):
         self.do_test_args("'a\"b'", 'a"b\0')
 
     @no_debug_info_test
+    @unittest2.expectedFailure
     def test_combined(self):
         self.do_test_args('"a b"c\'d e\'', 'a bcd e\0')
 
     @no_debug_info_test
+    @unittest2.expectedFailure
     def test_bare_single(self):
         self.do_test_args("a\\'b", "a'b\0")
 
     @expectedFailureWindows("http://llvm.org/pr24557")
     @no_debug_info_test
+    @unittest2.expectedFailure
     def test_bare_double(self):
         self.do_test_args('a\\"b', 'a"b\0')
 
