diff --git a/packages/Python/lldbsuite/test/functionalities/thread/create_after_attach/TestCreateAfterAttach.py b/packages/Python/lldbsuite/test/functionalities/thread/create_after_attach/TestCreateAfterAttach.py
index 9772543..12d9100 100644
--- a/packages/Python/lldbsuite/test/functionalities/thread/create_after_attach/TestCreateAfterAttach.py
+++ b/packages/Python/lldbsuite/test/functionalities/thread/create_after_attach/TestCreateAfterAttach.py
@@ -20,6 +20,7 @@ class CreateAfterAttachTestCase(TestBase):
                    # is implemented for FreeBSD.
     @skipIfWindows # Occasionally hangs on Windows, may be same as other issues.
     @skipIfiOSSimulator
+    @unittest2.expectedFailure
     def test_create_after_attach_with_popen(self):
         """Test thread creation after process attach."""
         self.build(dictionary=self.getBuildFlags(use_cpp11=False))
