diff --git a/packages/Python/lldbsuite/test/python_api/hello_world/TestHelloWorld.py b/packages/Python/lldbsuite/test/python_api/hello_world/TestHelloWorld.py
index 47c3ba1..31e80d4 100644
--- a/packages/Python/lldbsuite/test/python_api/hello_world/TestHelloWorld.py
+++ b/packages/Python/lldbsuite/test/python_api/hello_world/TestHelloWorld.py
@@ -75,7 +75,6 @@ class HelloWorldTestCase(TestBase):
 
     @add_test_categories(['pyapi'])
     @expectedFailureWindows("llvm.org/pr24600")
-    @expectedFailurei386("llvm.org/pr25338")
     @skipIfiOSSimulator
     def test_with_attach_to_process_with_id_api(self):
         """Create target, spawn a process, and attach to it with process id."""
@@ -105,7 +104,6 @@ class HelloWorldTestCase(TestBase):
 
     @add_test_categories(['pyapi'])
     @expectedFailureWindows("llvm.org/pr24600")
-    @expectedFailurei386("llvm.org/pr25338")
     @skipIfiOSSimulator
     def test_with_attach_to_process_with_name_api(self):
         """Create target, spawn a process, and attach to it with process name."""
