diff --git a/test/functionalities/register/TestRegisters.py b/test/functionalities/register/TestRegisters.py
index b78a254..6088dbe 100644
--- a/test/functionalities/register/TestRegisters.py
+++ b/test/functionalities/register/TestRegisters.py
@@ -21,6 +21,7 @@ class RegisterCommandsTestCase(TestBase):
         self.dbg.GetSelectedTarget().GetProcess().Destroy()
         TestBase.tearDown(self)
 
+    @unittest2.expectedFailure
     def test_register_commands(self):
         """Test commands related to registers, in particular vector registers."""
         if not self.getArchitecture() in ['amd64', 'i386', 'x86_64']:
@@ -28,6 +29,7 @@ class RegisterCommandsTestCase(TestBase):
         self.buildDefault()
         self.register_commands()
 
+    @unittest2.expectedFailure
     def test_fp_register_write(self):
         """Test commands that write to registers, in particular floating-point registers."""
         if not self.getArchitecture() in ['amd64', 'i386', 'x86_64']:
diff --git a/test/tools/lldb-server/TestGdbRemoteExpeditedRegisters.py b/test/tools/lldb-server/TestGdbRemoteExpeditedRegisters.py
index 57189a9..3a816a3 100644
--- a/test/tools/lldb-server/TestGdbRemoteExpeditedRegisters.py
+++ b/test/tools/lldb-server/TestGdbRemoteExpeditedRegisters.py
@@ -64,6 +64,7 @@ class TestGdbRemoteExpeditedRegisters(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @dwarf_test
+    @unittest2.expectedFailure
     def test_stop_notification_contains_any_registers_llgs_dwarf(self):
         self.init_llgs_test()
         self.buildDwarf()
@@ -88,6 +89,7 @@ class TestGdbRemoteExpeditedRegisters(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @dwarf_test
+    @unittest2.expectedFailure
     def test_stop_notification_contains_no_duplicate_registers_llgs_dwarf(self):
         self.init_llgs_test()
         self.buildDwarf()
@@ -107,6 +109,7 @@ class TestGdbRemoteExpeditedRegisters(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @dwarf_test
+    @unittest2.expectedFailure
     def test_stop_notification_contains_pc_register_llgs_dwarf(self):
         self.init_llgs_test()
         self.buildDwarf()
@@ -126,6 +129,7 @@ class TestGdbRemoteExpeditedRegisters(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @dwarf_test
+    @unittest2.expectedFailure
     def test_stop_notification_contains_fp_register_llgs_dwarf(self):
         self.init_llgs_test()
         self.buildDwarf()
@@ -145,6 +149,7 @@ class TestGdbRemoteExpeditedRegisters(gdbremote_testcase.GdbRemoteTestCaseBase):
 
     @llgs_test
     @dwarf_test
+    @unittest2.expectedFailure
     def test_stop_notification_contains_sp_register_llgs_dwarf(self):
         self.init_llgs_test()
         self.buildDwarf()
