diff --git a/test/functionalities/thread/thread_specific_break/TestThreadSpecificBreakpoint.py b/test/functionalities/thread/thread_specific_break/TestThreadSpecificBreakpoint.py
index 3faed3f..29981f0 100644
--- a/test/functionalities/thread/thread_specific_break/TestThreadSpecificBreakpoint.py
+++ b/test/functionalities/thread/thread_specific_break/TestThreadSpecificBreakpoint.py
@@ -24,6 +24,7 @@ class ThreadSpecificBreakTestCase(TestBase):
     @python_api_test
     @dwarf_test
     @expectedFlakeyLinux # this test fails 6/100 dosep runs
+    @skipIfLinux # usually hangs on ds2
     def test_with_dwarf_python(self):
         """Test that we obey thread conditioned breakpoints."""
         self.buildDwarf()
