diff --git a/test/dotest.py b/test/dotest.py
index fcfa1f7..2ba9eb3 100755
--- a/test/dotest.py
+++ b/test/dotest.py
@@ -635,7 +635,7 @@ def parseOptionsAndInitTestdirs():
         if platform_system == 'Darwin' and args.apple_sdk:
             compilers = [commands.getoutput('xcrun -sdk "%s" -find clang 2> /dev/null' % (args.apple_sdk))]
         else:
-            compilers = ['clang']
+            compilers = ['gcc']
 
     if args.channels:
         lldbtest_config.channels = args.channels
