diff --git a/packages/Python/lldbsuite/test/lldbtest.py b/packages/Python/lldbsuite/test/lldbtest.py
index de8f57f..9330406 100644
--- a/packages/Python/lldbsuite/test/lldbtest.py
+++ b/packages/Python/lldbsuite/test/lldbtest.py
@@ -976,13 +976,10 @@ def skipUnlessGoInstalled(func):
     return wrapper
 
 def getPlatform():
-    """Returns the target platform which the tests are running on."""
-    platform = lldb.DBG.GetSelectedPlatform().GetTriple().split('-')[2]
-    if platform.startswith('freebsd'):
-        platform = 'freebsd'
-    elif platform.startswith('netbsd'):
-        platform = 'netbsd'
-    return platform
+    # For some reason this likes to crash, so let's hack it
+    if sys.platform == 'darwin':
+        return 'darwin'
+    return 'linux'
 
 def getHostPlatform():
     """Returns the host platform running the test suite."""
